### ISSUE TITLE
Create a stub mapping for truncated nodes

### ISSUE THREAD

**Issue by kolesnikovae** [grafana/pyroscope#3196]
When a pprof profile is truncated, we replace the removed nodes with a shared "stub" location (named "other") that references the default mapping (ID == 1), which may lead to ambiguity. Instead, a shared "stub" mapping should be created (probably, even if the profile already has a "stub" mapping created at ingestion).

**Comment by simonswine** [comment]
Yes I do agree, esp. once we want to attach more information (OTEL spec) to the mapping we should be careful what belongs together or not.

**Comment by kolesnikovae** [comment]
@marc It might be crucial for read-path symbolication (e.g., #4203)
